File: src/spark/spark-script/src/main/scala/Sort.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object Sort {
    def main(args: Array[String]) {
        val conf = new SparkConf().setAppName("SortApp")
        val sc = new SparkContext(conf)
        val text_file = sc.textFile("/user/wangdongxu/input/file1.txt")
        val sortedCount = text_file.flatMap(line => line.split(" "))
            .map(word => (word, 1)) 
            .reduceByKey{case (x, y) => x + y}
            .map(_.swap)
            .sortByKey()

        val output = sortedCount.collect()
        for(i <- 0 until output.length)
            println(output(i))
    }
}

File: src/spark/spark-script/sort.py
#!/usr/bin/env python
from __future__ import print_function
from pyspark import SparkConf, SparkContext

if __name__ == "__main__":
    conf = (SparkConf()
             .setMaster("local")
             .setAppName("My app")
             .set("spark.executor.memory", "1g"))
    sc = SparkContext(conf = conf)
    # the input file is:
    # spark hadoop hive spark spark hadoop
    text_file = sc.textFile("/user/wangdongxu/input/file1.txt")
    #sortedCount = text_file.flatMap(lambda x: x.split(' ')) \
    #    .map(lambda x: (x, 1)) \
    #    .groupByKey()
    sortedCount = text_file.flatMap(lambda x: x.split(' ')) \
            .map(lambda word: (word, 1)) \
            .reduceByKey(lambda v1, v2: v1+v2) \
            .map(lambda (k,v) : (v,k)) \
            .sortByKey()
    output = sortedCount.collect()
    for (item, count) in output:
        print(item, count)

    # output:
    # 1 hive 
    # 2 hadoop
    # 3 spark
